weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx

```diff
@@ -0,0 +1,299 @@
+import {Box, Popover} from '@mui/material';
+import {
+  GridApiPro,
+  GridColumnVisibilityModel,
+  GridFilterModel,
+  GridSortModel,
+} from '@mui/x-data-grid-pro';
+import {Button} from '@wandb/weave/components/Button';
+import {Checkbox} from '@wandb/weave/components/Checkbox/Checkbox';
+import {TextField} from '@wandb/weave/components/Form/TextField';
+import {Icon, IconName} from '@wandb/weave/components/Icon';
+import {IconOnlyPill} from '@wandb/weave/components/Tag';
+import {Tailwind} from '@wandb/weave/components/Tailwind';
+import {maybePluralize} from '@wandb/weave/core/util/string';
+import React, {FC, useRef, useState} from 'react';
+
+import {useWFHooks} from '../wfReactInterface/context';
+import {ContentType} from '../wfReactInterface/traceServerClientTypes';
+import {WFHighLevelCallFilter} from './callsTableFilter';
+import {useDownloadFilterSortby} from './callsTableQuery';
+
+const MAX_EXPORT = 100_000;
+
+type SelectionState = {
+  allChecked?: boolean;
+  selectedChecked?: boolean;
+  limitChecked?: boolean;
+  limit?: string | null;
+  exportOption: ContentTypeOption;
+};
+
+type ContentTypeOption = {
+  value: ContentType | 'python' | 'curl';
+  text: string;
+  disabled?: boolean;
+};
+
+export const ExportSelector = ({
+  selectedCalls,
+  tableRef,
+  numTotalCalls,
+  columnVisibilityModel,
+  disabled,
+  callQueryParams,
+  rightmostButton = false,
+}: {
+  selectedCalls: string[];
+  tableRef: React.MutableRefObject<GridApiPro>;
+  numTotalCalls: number;
+  columnVisibilityModel: GridColumnVisibilityModel | undefined;
+  callQueryParams: {
+    entity: string;
+    project: string;
+    filter: WFHighLevelCallFilter;
+    gridFilter: GridFilterModel;
+    gridSort?: GridSortModel;
+  };
+  disabled: boolean;
+  rightmostButton?: boolean;
+}) => {
+  const defaultSelectionState = {
+    exportOption: {value: ContentType.jsonl, text: 'jsonl'},
+    allChecked: true,
+  };
+  const [selectionState, setSelectionState] = useState<SelectionState>(
+    defaultSelectionState
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const onClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(anchorEl ? null : ref.current);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popper' : undefined;
+  const cols = columnVisibilityModel ?? {};
```

Comment:
this doesn't work, the visibility model cols are all false unless the user explicitly adds them. Fix